nc_perf/tst_files3.c

```diff
@@ -138,7 +139,7 @@ int dump_hdf_file(const float *data, int docompression)
       for (start[1] = 0; start[1] < Y_LEN; start[1]++)
       {
 	 if (H5Sselect_hyperslab(file_spaceid, H5S_SELECT_SET, start, NULL,
-				 count, NULL) < 0) ERR_RET;
+				 ones, count) < 0) ERR_RET;
```

Comment:
So we are replacing count, `{1,1,128}` with ones, `{1,1,1}`, and passing `{1,1,128}` as the 6th argument to `H5Select_hyperslabs()` instead of `NULL`.  As per the assertion in the conversation, `NULL` would be treated as `{1,1,1}` when passed to `H5Select_hyperslabs()`.  This difference may not matter (I need to go refer to the HDF5 API documentation); given that the tests still pass, it's probably safe bet that it doesn't matter.  However, I just wanted to raise this inconsistency when it was noticed.  @wkliao, is this intentional/inconsequential?

Reply:
> So we are replacing count, `{1,1,128}` with ones, `{1,1,1}`, and passing `{1,1,128}` as the 6th argument to `H5Select_hyperslabs()` instead of `NULL`. As per the assertion in the conversation, `NULL` would be treated as `{1,1,1}` when passed to `H5Select_hyperslabs()`. This difference may not matter (I need to go refer to the HDF5 API documentation); given that the tests still pass, it's probably a safe bet that it doesn't matter. However, I just wanted to raise this inconsistency when it was noticed. @wkliao, is this intentional/inconsequential?

The use of `count, NULL` in the NetCDF current implementation does not cause any error. My intension of this PR is to point out the correct usage of `H5Sselect_hyperslabs()`. Please refer to the [HDF5 reference manual](https://support.hdfgroup.org/HDF5/doc/RM/H5S/H5Sselect_hyperslab.htm). There is also an example that helps understand arguments count and block, quoted below
```
For example, consider a 2-dimensional dataspace with hyperslab
selection settings as follows: the start offset is specified as [1,1],
stride is [4,4], count is [3,7], and block is [2,2]. In C, these settings
will specify a hyperslab consisting of 21 2x2 blocks of array elements
starting with location (1,1) with the selected blocks at locations
(1,1), (5,1), (9,1), (1,5), (5,5), etc.; 
```

My concern is for performance only. Even though the outcomes are the
same, the internal HDF5 implementation will treat each block as a single
element block if NULL is used. For example, a subarray of size 5x3 will be
stored as 15 blocks of size 1x1. If this PR is used, then the subarray will be
stored as 1 block of size 5x3.